tests/v2/validation/workloads/workload_test.go

```diff
@@ -192,6 +200,159 @@ func (w *WorkloadTestSuite) TestWorkloadStatefulset() {
+		FieldSelector: "metadata.name=" + redisDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, redisDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err = deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, redisDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1", "2"}, revisions)
+
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "redis")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	containerName = namegen.AppendRandomString("updatetestcontainertwo")
+	newContainerTemplate = workloads.NewContainer(containerName,
+		"ubuntu",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	newContainerTemplate.TTY = true
+	newContainerTemplate.Stdin = true
+	ubuntuDeployment := redisDeployment.DeepCopy()
+	ubuntuDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	_, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+
+	err = charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + ubuntuDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err = deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1", "2", "3"}, revisions)
+
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "ubuntu")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	logRollback, err := deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, nginxDeployment, 1)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
+
+	err = charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + nginxDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, nginxDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err = deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, nginxDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"2", "3", "4"}, revisions)
+
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "nginx")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	logRollback, err = deployment.RollbackDeployment(w.client, w.cluster.ID, namespace.Name, redisDeployment, 2)
+	require.NoError(w.T(), err)
+	require.NotEmpty(w.T(), logRollback)
```

Comment:
just pointing out that, the deployment isn't used in rollbackDeployment, just its name. so this is sort of confusing just reading through the code when using copied versions of the deployment vs. just using the same deployment / deploymentName and requesting to rollback to specific revisions